File: eventemitter.pl
use v5.14;

use Mojo::EventEmitter;


package Cat {
    use Mojo::Base 'Mojo::EventEmitter';
    sub poke {
        my ($self) = @_;
        $self->emit(roar => 3);
    }
}


my $tiger = Cat->new();

$tiger->on(roar => sub {
    my ($name, $times) = @_;
    for (1 .. $times) {
        say 'RAWR!';
    }
});

$tiger->poke();

File: eventemitter.py
from __future__ import print_function

from Pyjo.EventEmitter import *


class Cat(Pyjo_EventEmitter):
    def poke(self):
        self.emit('roar', 3)


def roar(name, times):
    for _ in range(0, times):
        print('RAWR!')


tiger = Cat()
tiger.on('roar', roar)
tiger.poke()
